.github/workflows/cypress-Run-all-tests-no-cache.yml

```diff
@@ -69,32 +83,50 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v2
         with:
-          node-version: 14
+          node-version: 16
 
+      # On push o r when we specify we don't want to send the tests to Dashboard through workflow dispatch
+      # we set the environment variables dynamically to empty in order to avoid
+      # recording the test execution to Cypress Dashboard.
+      - name: Set Dashboard flags to empty when push or worflow_dispatch
+        if: ${{ github.event_name == 'push' || github.event.inputs.cyDashboard == 'No' }}
+        run: |
+          echo "CY_RECORD_KEY=" >> $GITHUB_ENV
+          echo "CY_RECORD_FLAG=" >> $GITHUB_ENV
+
+      # We define the environment variables to send the test execution to the Cypress Dashboard
+      # on every pull request or when specifying to record the tests on workflow dispatch
+      - name: Set Dashboard flags when pull_request or worflow_dispatch
+        if: ${{ github.event_name == 'pull_request' || github.event.inputs.cyDashboard == 'Yes' || github.event_name == 'schedule' }}
+        run: |
+          echo "CY_RECORD_KEY=${{ secrets.CYPRESS_RECORD_KEY }}" >> $GITHUB_ENV
+          echo "CY_RECORD_FLAG=-- --record --key " >> $GITHUB_ENV
+     
       - name: Run tests by browser
         uses: cypress-io/github-action@v2
         timeout-minutes: 10
         with:
           # 'build' starts the defauld demo 
           build: npm run build
-          # 'test:ci' runs tests over Docker image for build context 
-          command: npm run test:ci
-          record: false
-          parallel: false
+          # 'test:ci' runs tests over Docker image for build context
+          # we also send the Cypress Dashboard record key dynamically
+          command: npm run test:ci ${{ env.CY_RECORD_FLAG }} ${{ env.CY_RECORD_KEY }}
+          record: true
```

Comment:
I have a doubt, if you don' mind.

Shouldn't be better to indicate whether we send `record` data to Cypress with a variable as calculated on the `Set Dashboard flags when pull_request or worflow_dispatch` step?

Otherwise, `CY_RECORD_FLAG` and `CY_RECORD_KEY` wont' be initialized and thus not sending record data to Cypress, right?

Reply:
Since we want to send data to cypress dashboard on some specific cases, we initialize those variables only when we want to send the data to the dashboard, so the situations where they're not initialized, hence they don't send data to the dashboard, are working as they should.